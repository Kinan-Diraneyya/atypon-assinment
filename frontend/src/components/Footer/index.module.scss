@use '../../utilities/utilities.scss' as u;

.footer {
    background-color: #4c4c4c;

    .text {
        @include u.content-area;
        padding-top: 0.8rem;
        padding-bottom: 0.8rem;
        color: white;
        font-size: 0.8rem;
    }
}