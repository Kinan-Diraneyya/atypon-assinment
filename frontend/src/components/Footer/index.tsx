import style from './index.module.scss';

/**
 * Site footer.
 * 
 * Renders a ribbon with a few details about the site.
 * 
 * @returns {JSX.Element} The Footer component.
 */
export default function Footer() {
    return (
        <footer className={style.footer}>
            <small className={style.text}>Created on 21/07/2024 for Atypon online assignment</small>
        </footer>
    );
}