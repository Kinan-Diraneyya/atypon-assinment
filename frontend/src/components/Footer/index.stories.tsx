import React from 'react';
import Footer from './index';
import { Meta, StoryObj } from '@storybook/react';

const meta: Meta<typeof Footer> = {
    component: Footer,
  };
   
  export default meta;
  type Story = StoryObj<typeof Footer>;
   
  export const Main: Story = {
    args: {}
  };