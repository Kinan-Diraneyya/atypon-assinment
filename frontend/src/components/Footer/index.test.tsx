import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Footer from './index';

describe('Footer Component', () => {
  test('renders without crashing', () => {
    render(<Footer />);
    const logoElement = screen.getByText('Created on 21/07/2024 for Atypon online assignment');
    expect(logoElement).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });
});