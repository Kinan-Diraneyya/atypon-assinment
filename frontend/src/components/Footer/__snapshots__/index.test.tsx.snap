// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component matches snapshot 1`] = `
<DocumentFragment>
  <footer
    class="footer"
  >
    <small
      class="text"
    >
      Created on 21/07/2024 for Atypon online assignment
    </small>
  </footer>
</DocumentFragment>
`;
